handlers.go

```diff
@@ -432,6 +443,130 @@ func (s *Server) handleStats(c echo.Context, u *User) error {
 	return c.JSON(200, out)
 }
 
+// handlePeeringPeersAdd godoc
+// @Summary      Add peers on Peering Service
+// @Description  This endpoint can be used to add a Peer from the Peering Service
+func (s *Server) handlePeeringPeersAdd(c echo.Context) error {
+	var params []peering.PeeringPeer
+	if err := c.Bind(&params); err != nil {
+		return err
+	}
+
+	for _, peerParam := range params {
+
+		var multiAddrs []multiaddr.Multiaddr
+		for _, addr := range peerParam.Addrs {
+			a, err := multiaddr.NewMultiaddr(addr)
+			if err != nil {
+				return &util.HttpError{
+					Code:    400,
+					Message: util.ERR_PEERING_PEERS_ADD_ERROR,
+				}
+			}
+			multiAddrs = append(multiAddrs, a)
+		}
+
+		peerParamId, _ := peer.Decode(peerParam.ID)
+		//	add the peer(s)
+		s.Node.Peering.AddPeer(
```

Comment:
why not build/verify the multiaddr before adding it for peering? if it fails halfway, it means partial adding. Except this it your intention

Reply:
I meant the complete peer sent for the request - if there are multiple peers, 1 fails and 1 pass - it adds 1, yet returns an error for the request. So my question is, is that intended?

Reply:
if one of the address is invalid, then the entire request should be invalid. It would be best for us to impose an absolute correctness of the addresses and inform the consumer if one of them is invalid.

Reply:
@alvin-reyes on line 479, it is not all or nothing - which is what I am pointing at

Reply:
Yes. That's why we return a json response immediately if one of the addrs is invalid.

Line 466 - 475
```
for _, addr := range peerParam.Addrs {
			a, err := multiaddr.NewMultiaddr(addr)
			if err != nil {
				log.Errorf("handlePeeringPeersAdd error: %s", err)
				return c.JSON(http.StatusBadRequest,
					util.PeeringPeerAddMessage{
						"Adding Peer(s) on Peering failed, the addr is invalid" + addr, params})
			}
			multiAddrs = append(multiAddrs, a)
		}
```

If one of them is invalid, the entire request to add will be cancelled with StatusBadRequest response. It'll also show the message and list of invalid addrs.

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/4479171/173366223-4bc579d6-b3c7-4e4c-b667-7a75f175fec9.png">

Reply:
you are looking at per addr level. I am talking about per peer level

Reply:
If one of the peers are invalid, then we reject the entire request and raise the error similar to how we handle peer addrs level.

<img width="1133" alt="image" src="https://user-images.githubusercontent.com/4479171/173387619-49c7943b-43d3-4ab5-98aa-bc5cef1fa5c3.png">

It make sense to reject the entire request if one of the peers or peer.addrs are invalid.